### PR TITLE
Hw hibernate user service

### PR DESCRIPTION
**Pull request by OlehMytiai**

### CODE REVIEW

**Review by NazariiYatsiuk — APPROVED** [review]
I see no problems

File: src/main/java/mate/academy/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        if (email == null || password == null
+                || email.isBlank() || password.isBlank()
+                || userService.findByEmail(email).isPresent()) {
+            throw new RegistrationException("Incorrect email or password");
+        }
```
Comment (KostyaNikiforov) [inline src/main/java/mate/academy/security/AuthenticationServiceImpl.java:23]: Just for the future, it is a good practice to move big conditions to a separate method

File: src/main/java/mate/academy/util/HashUtil.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    private static final String CRYPTO_ALGORITHM = "SHA-512";
+
+    private HashUtil() {
+
```
Comment (KostyaNikiforov) [inline src/main/java/mate/academy/util/HashUtil.java:11]: Empty line

File: src/main/java/mate/academy/util/HashUtil.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+
+    public static byte[] getSalt() {
+        SecureRandom secureRandom = new SecureRandom();
+        byte[] salt = new byte[16];
```
Comment (KostyaNikiforov) [inline src/main/java/mate/academy/util/HashUtil.java:16]: Let's make this value constant too

**Review by KostyaNikiforov — CHANGES_REQUESTED** [review]

**Review by KostyaNikiforov — APPROVED** [review]

**Pull request closed without merging**